README.md

```diff
@@ -1 +1,172 @@
+
+```python
+transform = transforms.Compose([
+    transforms.Resize(256),
+    transforms.CenterCrop(250),
+    transforms.ToTensor()
+])
+
+whole_dataset = ImageFolder(path, transform=transform)
+
+batch_size = 100
+num_workers = 64
+indices = list(range(len(data)))
+np.random.shuffle(indices)
+train_idx = indices[:num]
+test_idx = indices[num:num+num]
+
+train_sampler = SubsetRandomSampler(train_idx)
+train_loader = DataLoader(data, sampler=train_sampler, batch_size=batch_size, num_workers=num_workers)
+```
+
+### Training a Model
+Loop over the dataset, and train the model by stepping the optimizer
+
+```python
+device = torch.device(0)
+net = models.resnet18(pretrained=False)
+model = net.to(device)
+device_ids = [0]
+
+criterion = nn.CrossEntropyLoss().cuda()
+lr = 0.001
+optimizer = optim.SGD(model.parameters(), lr=lr, momentum=0.9)
+count = 0
+for epoch in range(n_epochs):
+    model.train()  # Set model to training mode
+    for inputs, labels in train_loader:
+        dt = datetime.datetime.now().isoformat()
+        inputs = inputs.to(device)
+        labels = labels.to(device)
+        outputs = model(inputs)
+        _, preds = torch.max(outputs, 1)
+        loss = criterion(outputs, labels)
+
+        # zero the parameter gradients
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+        count += 1
+```
+
+## Now on Dask
+
+with dask_pytorch and PyTorch distributed data parallel, we can train on multiple workers as follows:
+
+### Loading Data
+Load the dataset from S3, and explicitly set the multiprocessing context (Dask defaults to spawn, but pytorch is generally configured to use fork)
+
+```python
+from dask_pytorch.data import S3ImageFolder
+
+whole_dataset = S3ImageFolder(bucket, prefix, transform=transform)
+train_loader = torch.utils.data.DataLoader(
+    whole_dataset, sampler=train_sampler, batch_size=batch_size, num_workers=num_workers, multiprocessing_context=mp.get_context('fork')
+)
+```
+
+### Training in Parallel
+
+Wrap the training loop in a function (and add metrics logging.  Not necessary, but very useful).  Convert the model into a `DDP` model which knows how to sync gradients together across workers.
+
+```python
+import uuid
+import pickle
+import logging
+import json
+
+
+key = uuid.uuid4().hex
+rh = DaskResultsHandler(key)
+
+def run_transfer_learning(bucket, prefix, samplesize, n_epochs):
+    worker_rank = int(dist.get_rank())
+    device = torch.device(0)
+    net = models.resnet18(pretrained=False)
+    model = net.to(device)
+    model = DDP(model, device_ids=[0])
+
+    criterion = nn.CrossEntropyLoss().cuda()
+    lr = 0.001 * dist.get_world_size()
+    optimizer = optim.SGD(model.parameters(), lr=lr, momentum=0.9)
+    whole_dataset = BOTOS3ImageFolder(bucket, prefix, transform=transform)
+    train_loader = torch.utils.data.DataLoader(
+        whole_dataset,
+        sampler=train_sampler,
+        batch_size=batch_size,
```

Comment:
I think the working script was passing a dict for some of these arguments in, but since we want this to be a simplified example I'll just make these arguments to the function.